providers/system_proxy.rb

```diff
@@ -91,10 +92,14 @@
                 fe = Chef::Util::FileEdit.new("/etc/environment")
                 fe.search_file_replace_line(/HTTP_PROXY/i,"HTTP_PROXY=\"#{global_settings['http_proxy']}:#{global_settings['http_proxy_port']}\"")
                 fe.search_file_replace_line(/HTTPS_PROXY/i,"HTTPS_PROXY=\"#{global_settings['https_proxy']}:#{global_settings['https_proxy_port']}\"")
+                fe.write_file
```

Comment:
Chef (write_file method) only save the last file_replace_line. It's not a documented behaviour but you can find the same problem on forums, for example:

https://discourse.chef.io/t/inserting-multiple-lines-through-chef-util-fileedit/3676